### PR TITLE
Add oraclelinux:8-slim as base image for GraalVM-CE

### PR DESCRIPTION
**Pull request by aboullaite**
This PR is based on the discussion #1528 about using OL8 as base image for GraalVM-CE

Signed-off-by: Mohammed Aboullaite <aboullaite.mohammed@gmail.com>

### CODE REVIEW

**Comment by Djelibeybi** [comment]
As per the Contributiong Guide, could you pleae review and return a signed Oracle Contributor Agreement: https://www.oracle.com/technetwork/community/oca-486395.html

If you have already submitted one, let me know and I'll look into it.

**Comment by aboullaite** [comment]
@Djelibeybi i did a couple of days ago.

**Comment by Djelibeybi** [comment]
Thanks. I've just sent you an email about your CLA submission (there's a checkbox missing from the form).

**Comment by aboullaite** [comment]
My bad! Fixed and sent.

**Comment by Djelibeybi** [comment]
OCA verified. @mzachh please go ahead and review this PR.

File: GraalVM/CE/Dockerfile.ol8 [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# Note: If you are behind a web proxy, set the build variables for the build:
+#       E.g.:  docker build --build-arg "https_proxy=..." --build-arg "http_proxy=..." --build-arg "no_proxy=..." ...
+
+ARG GRAALVM_VERSION=20.0.0
+ARG JAVA_VERSION=java8
+ARG GRAALVM_PKG=https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-$GRAALVM_VERSION/graalvm-ce-$JAVA_VERSION-linux-amd64-$GRAALVM_VERSION.tar.gz 
+ENV LANG=en_US.UTF-8 JAVA_HOME=/opt/graalvm-ce-$JAVA_VERSION-$GRAALVM_VERSION/
+
+RUN microdnf update -y oraclelinux-release-el8 \
+    && microdnf --enablerepo ol8_appstream install -y bzip2-devel ed gcc gcc-c++ gcc-gfortran gzip file fontconfig less libcurl-devel make openssl openssl-devel readline-devel tar vi which xz-devel zlib-devel \
```
Comment (Djelibeybi) [inline GraalVM/CE/Dockerfile.ol8:17]: The `ol8_appstream` repo is enbled by default, so you don't need to specify it here.

File: GraalVM/CE/Dockerfile.ol8 [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# LICENSE UPL 1.0
+#
+# Copyright (c) 2015 Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline GraalVM/CE/Dockerfile.ol8:3]: The date should be 2020 and please remove "All rights reserved." too. You also need to add a second line, as follows: 
```
Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
```

File: GraalVM/CE/Dockerfile.ol8 [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# LICENSE UPL 1.0
```
Comment (Djelibeybi) [inline GraalVM/CE/Dockerfile.ol8:1]: This line can be removed.

File: GraalVM/CE/Dockerfile.ol8 [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+ARG GRAALVM_VERSION=20.0.0
+ARG JAVA_VERSION=java8
+ARG GRAALVM_PKG=https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-$GRAALVM_VERSION/graalvm-ce-$JAVA_VERSION-linux-amd64-$GRAALVM_VERSION.tar.gz 
+ENV LANG=en_US.UTF-8 JAVA_HOME=/opt/graalvm-ce-$JAVA_VERSION-$GRAALVM_VERSION/
+
+RUN microdnf update -y oraclelinux-release-el8 \
+    && microdnf --enablerepo ol8_appstream install -y bzip2-devel ed gcc gcc-c++ gcc-gfortran gzip file fontconfig less libcurl-devel make openssl openssl-devel readline-devel tar vi which xz-devel zlib-devel \
+    && microdnf --enablerepo ol8_codeready_builder install -y glibc-static libstdc++ libstdc++-devel libstdc++-static zlib-static \
+    && microdnf clean all && rm -rf /var/cache/yum
```
Comment (Djelibeybi) [inline GraalVM/CE/Dockerfile.ol8:19]: `microdnf clean all` is sufficient to remove all locally generated metadata.

File: GraalVM/CE/Dockerfile.ol8-java11 [hunk 5]
```diff
@@ -0,0 +1,44 @@
+# LICENSE UPL 1.0
```
Comment (Djelibeybi) [inline GraalVM/CE/Dockerfile.ol8-java11:1]: This line can be removed.

File: GraalVM/CE/Dockerfile.ol8-java11 [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+#
+# Copyright (c) 2015 Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline GraalVM/CE/Dockerfile.ol8-java11:3]: The date should be 2020 and please remove "All rights reserved." too. You also need to add a second line, as follows: 
```
Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
```

File: GraalVM/CE/Dockerfile.ol8-java11 [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+ARG GRAALVM_VERSION=20.0.0
+ARG JAVA_VERSION=java11
+ARG GRAALVM_PKG=https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-$GRAALVM_VERSION/graalvm-ce-$JAVA_VERSION-GRAALVM_ARCH-$GRAALVM_VERSION.tar.gz 
+ARG TARGETPLATFORM
+
+ENV LANG=en_US.UTF-8 JAVA_HOME=/opt/graalvm-ce-$JAVA_VERSION-$GRAALVM_VERSION/
+
+RUN microdnf update -y oraclelinux-release-el8 \
+    && microdnf --enablerepo ol8_appstream install -y bzip2-devel ed gcc gcc-c++ gcc-gfortran gzip file fontconfig less libcurl-devel make openssl openssl-devel readline-devel tar vi which xz-devel zlib-devel \
```
Comment (Djelibeybi) [inline GraalVM/CE/Dockerfile.ol8-java11:19]: The `ol8_appstream` repo is enabled by default, so it doesn't need to be specified.

File: GraalVM/CE/Dockerfile.ol8-java11 [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    && microdnf --enablerepo ol8_codeready_builder install -y glibc-static libstdc++ libstdc++-devel libstdc++-static zlib-static \
+    && microdnf clean all && rm -rf /var/cache/yum
```
Comment (Djelibeybi) [inline GraalVM/CE/Dockerfile.ol8-java11:21]: `microdnf clean all` is sufficient to remove locally generated metadata

**Comment by aboullaite** [comment]
Made another commit with the requested changes.

**Comment by mzachh** [comment]
Thank you for your contribution!

I will verify the Dockerfile in the next days.

File: GraalVM/CE/Dockerfile.ol8 [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+# Note: If you are behind a web proxy, set the build variables for the build:
+#       E.g.:  docker build --build-arg "https_proxy=..." --build-arg "http_proxy=..." --build-arg "no_proxy=..." ...
+
+ARG GRAALVM_VERSION=20.0.0
+ARG JAVA_VERSION=java8
+ARG GRAALVM_PKG=https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-$GRAALVM_VERSION/graalvm-ce-$JAVA_VERSION-linux-amd64-$GRAALVM_VERSION.tar.gz 
+ENV LANG=en_US.UTF-8 JAVA_HOME=/opt/graalvm-ce-$JAVA_VERSION-$GRAALVM_VERSION/
+
+RUN microdnf update -y oraclelinux-release-el8 \
+    && microdnf install -y bzip2-devel ed gcc gcc-c++ gcc-gfortran gzip file fontconfig less libcurl-devel make openssl openssl-devel readline-devel tar vi which xz-devel zlib-devel \
```
Comment (mzachh) [inline GraalVM/CE/Dockerfile.ol8:15]: Please could you also add the `findutils` package, the gu-wrapper requires the `find` utility.

Comment (Djelibeybi) [reply GraalVM/CE/Dockerfile.ol8:15]: `microdnf` also doesn't need the `-y` parameter, but it's not a showstopper. If you're editing the file anyway, you may as well remove them though.

Comment (aboullaite) [reply GraalVM/CE/Dockerfile.ol8]: @mzachh @Djelibeybi Both changes are committed

Comment (Djelibeybi) [reply GraalVM/CE/Dockerfile.ol8]: What's the reason for having two calls to `microdnf`? Why not just install everything at once, i.e. what am I missing here? :)

Comment (aboullaite) [reply GraalVM/CE/Dockerfile.ol8]: You're right! I made the requested changes

**Comment by Djelibeybi** [comment]
Thanks for your contribution!

**Pull request merged**